commit 5996b44f90a0bebccbc8d4e70aad12d9cf2cc9fb
Author: Paul Cercueil <paul@crapouillou.net>
Date:   Thu Jan 16 23:10:06 2014 +0100

    Allow exiting the application by pressing B

diff --git a/build/wificonfig.py b/build/wificonfig.py
index 25c2b1d..c9ac1bc 100755
--- a/build/wificonfig.py
+++ b/build/wificonfig.py
@@ -1644,6 +1644,9 @@ if __name__ == "__main__":
 						active_menu = to_menu("main")
 						del uniq
 						redraw()
+					elif event.key == K_LALT:
+						pygame.display.quit()
+						sys.exit()
 				elif event.key == K_SPACE:
 					if active_menu == "saved":
 						if len(str(wirelessmenu.get_selected()[0])) > 16: